joomla/joomla-cms
# Module extensions are not uninstalling correctly on package install rollback

Pull Request for New Issue.

### Summary of Changes

The module installer adapter is not working correct when a install error exists and the system needs to rollback the changes.
This PR intends to fix that.

### Testing Instructions

Use latest staging and use this already made [pkg_weblinks.zip](https://github.com/joomla/joomla-cms/files/667676/pkg_weblinks.zip) corrupted test package (just added  `<folder>whatever</folder>` to plg_finder_weblinks manifest to produce an error)


- Reproduce the issue

  - Install the corrupted package you will get and error and install of the package should rollback
  - Check extensions manage, order by id, and check you have the module installed

- Test patch: Apply patch, install the package and you will see the module is now uninstalled correctly on rollback

### Documentation Changes Required

None

### More info

You can see in the other extension adapters that the var name shoudl be called `id`, not `extension_id`, exmaple for [plugins](https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/installer/adapter/plugin.php#L444) and [libraries](https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/installer/adapter/library.php#L287).

Also that is the `id` that is checked in the [installer abort method](https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/installer/installer.php#L408)
